packages/smooth_app/lib/database/dao_product.dart

```diff
@@ -102,7 +103,10 @@ class DaoProduct extends AbstractSqlDao implements BulkDeletable {
     final List<dynamic> insertParameters = <dynamic>[];
     for (final Product product in products) {
       insertParameters.add(product.barcode);
-      insertParameters.add(json.encode(product.toJson()));
+      // The data is compressed with gzip to save space.
+      final List<int> compressed =
+          gzip.encode(utf8.encode(jsonEncode(product.toJson())));
+      insertParameters.add(compressed);
```

Comment:
``` dart
    await db.execute('create table $TABLE_PRODUCT('
          // cf. https://www.sqlite.org/lang_conflict.html
          '$TABLE_PRODUCT_COLUMN_BARCODE TEXT PRIMARY KEY on conflict replace'
          ',$_TABLE_PRODUCT_COLUMN_JSON BLOB NOT NULL'
          ')');
    }
```
So something like this will work good as u said ryt?